src/main/java/com/ft/methodearticlemapper/transformation/DynamicContentXMLEventHandler.java

```diff
@@ -0,0 +1,65 @@
+package com.ft.methodearticlemapper.transformation;
+
+import com.ft.bodyprocessing.BodyProcessingContext;
+import com.ft.bodyprocessing.writer.BodyWriter;
+import com.ft.bodyprocessing.xml.eventhandlers.BaseXMLEventHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.xml.stream.XMLEventReader;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.events.StartElement;
+import java.util.HashMap;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+public class DynamicContentXMLEventHandler extends BaseXMLEventHandler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DynamicContentXMLEventHandler.class);
+
+    private static final String FT_CONTENT_TAG = "ft-content";
+
+    private DynamicContentXMLParser dynamicContentXMLParser;
+
+    public DynamicContentXMLEventHandler(DynamicContentXMLParser dynamicContentXMLParser) {
+        checkArgument(dynamicContentXMLParser != null, "slideshowXMLParser cannot be null");
+        this.dynamicContentXMLParser = dynamicContentXMLParser;
+    }
+
+    @Override
+    public void handleStartElementEvent(StartElement event,
+                                        XMLEventReader xmlEventReader,
+                                        BodyWriter eventWriter,
+                                        BodyProcessingContext bodyProcessingContext) throws XMLStreamException {
+        DynamicContentData dynamicContentData = dynamicContentXMLParser.parseElementData(event, xmlEventReader, bodyProcessingContext);
+        if (!dynamicContentData.isAllRequiredDataPresent()) {
+            return;
+        }
+
+        String apiHost = getApiHostFromContext(bodyProcessingContext);
+
+        HashMap<String, String> attributes = new HashMap<>();
+        attributes.put("type", "http://www.ft.com/ontology/content/DynamicContent");
```

Comment:
Can we extract constants?

Reply:
I don't feel like extracting this String as a constant adds any value since it's not used in multiple places.